router.go

```diff
@@ -305,22 +312,26 @@ func (r *Router) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		handle.ServeHTTP(w, req)
 		return
 	} else if req.Method != http.MethodConnect && path != "/" {
```

Comment:
It took me a minute to understand the logic here. I would probably split this out into its own top-level `if` and add comment(s) like:

```go
if handle, params := r.lookup(req.Method, path); handle != nil {
	// ...
	return
}

// No route matched the incoming request. Try any automatic fallbacks that are enabled.

// Redirect from (e.g.) `/foo/` to `/foo` (or vice-versa):
if r.RedirectTrailingSlash && path != "/" && req.Method != http.MethodConnect {
```